localstack-core/localstack/services/sns/filter.py

```diff
@@ -124,6 +125,10 @@ def _evaluate_condition(self, value, condition, field_exists: bool):
             return equal_ignore_case.lower() == value.lower()
         elif numeric_condition := condition.get("numeric"):
             return self._evaluate_numeric_condition(numeric_condition, value)
+        elif cidr := condition.get("cidr"):
+            ips = [str(ip) for ip in ipaddress.IPv4Network(cidr)]
+            return value in ips
```

Comment:
That seems super inefficient. :thinking: If the mask is `10.0.0.0/8` that would create a huge list. :eyes: 

If we are sure it will always be an ipv4 address we can maybe do that `return ipaddress.IPv4Address(value) in ipaddress.IPv4Network(cidr)` and something like `return ipaddress.ip_address(value) in ipaddress.ip_network(cidr)` would cover ipv6 also, I believe

Reply:
Right, yes. Well, we can't be sure the `value` is a valid ipv4 or v6. Actually, looking at the docs, it seems it should work with ipv6, so this doesn't really work either. I'll update `events` as well to add IPv6 support. 

I'll need to wrap this in a try/except block probably, and do what you proposed.

Reply:
time to port back the fix to `events` 😂 thanks, always pushing further! 💪 always thankful for your reviews!